api/providers/views.py

```diff
@@ -611,11 +643,100 @@ class RegistrationProviderSchemaList(JSONAPIBaseView, generics.ListAPIView, List
         return self.get_provider().schemas.get_latest_versions(request=self.request, invisible=True).filter(active=True)
 
     def get_queryset(self):
         return self.get_queryset_from_request()
+
+
+class RegistrationProviderRegistrationList(JSONAPIBaseView, generics.ListAPIView, ListFilterMixin, ProviderMixin):
+    provider_class = RegistrationProvider
+    permission_classes = (
+        drf_permissions.IsAuthenticated,
+        base_permissions.TokenHasScope,
+        MustBeModerator,
+    )
+
+    ordering = ('-created')
+
+    serializer_class = RegistrationSerializer
+
+    required_read_scopes = [CoreScopes.NODE_REGISTRATIONS_READ]
+    required_write_scopes = [CoreScopes.NULL]
+
+    view_category = 'registration-providers'
+    view_name = 'registrations-list'
+
+    def get_default_queryset(self):
+        provider = self.get_provider()
+
+        return Registration.objects.filter(
+            provider=provider,
+            registration_approval__state=RegistrationApproval.APPROVED,
+        ).annotate(machine_state=F(provider.MODERATION_MACHINE_STATE_RELATION))
+
+    # overrides ListAPIView
+    def get_queryset(self):
+        return self.get_queryset_from_request()
+
+    # overrides APIView
+    def get_renderer_context(self):
+        context = super().get_renderer_context()
+        if is_truthy(self.request.query_params.get('meta[reviews_state_counts]', False)):
+            provider = self.get_provider()
+            context['meta'] = {
+                'reviews_state_counts': provider.get_reviewable_state_counts(),
+            }
+        return context
+
+
+class RegistrationProviderRequestList(JSONAPIBaseView, generics.ListAPIView, ListFilterMixin, ProviderMixin):
+    provider_class = RegistrationProvider
+    permission_classes = (
+        drf_permissions.IsAuthenticated,
+        base_permissions.TokenHasScope,
+        MustBeModerator,
+    )
+    view_category = 'requests'
+    view_name = 'registration-provider-request-list'
+
+    required_read_scopes = [CoreScopes.REGISTRATION_REQUESTS_READ]
+    required_write_scopes = [CoreScopes.NULL]
+
+    serializer_class = RegistrationRequestSerializer
+
+    def get_default_queryset(self):
+        return NodeRequest.objects.filter(
+            target__provider_id=self.get_provider().id,
+            target__deleted__isnull=True,
+        )
+
+    def get_queryset(self):
+        return self.get_queryset_from_request()
+
+
+class RegistrationProviderActionList(JSONAPIBaseView, generics.ListAPIView, ListFilterMixin, ProviderMixin):
+    provider_class = RegistrationProvider
+
+    permission_classes = (
+        drf_permissions.IsAuthenticated,
+        base_permissions.TokenHasScope,
+        MustBeModerator,
+    )
+    view_category = 'actions'
+    view_name = 'registration-provider-action-list'
+
+    required_read_scopes = [CoreScopes.ACTIONS_READ]
+    required_write_scopes = [CoreScopes.NULL]
+
+    serializer_class = NodeRequestActionSerializer
```

Comment:
If the `serializer_class` is `NodeRequestActionSerializer`, that means the `type` for this resource is `node-request-actions`. Does that mean every such action is related to a request?